lib/doorkeeper/grape/helpers.rb

```diff
@@ -0,0 +1,59 @@
+module Doorkeeper
+  module Grape
+    module Helpers
+      extend ActiveSupport::Concern
+
+      included do
+        helpers do
+          def doorkeeper_token
+            @_doorkeeper_token ||= OAuth::Token.authenticate request, *Doorkeeper.configuration.access_token_methods
+          end
+
+          def valid_doorkeeper_token?(*scopes)
+            doorkeeper_token && doorkeeper_token.acceptable?(scopes)
+          end
+
+          # endpoint specific scopes > parameter scopes > default scopes
+          def doorkeeper_authorize!(*scopes)
```

Comment:
Assignment Branch Condition size for doorkeeper_authorize! is too high. [36.07/15]<br>Perceived complexity for doorkeeper_authorize! is too high. [10/7]